code/modules/reagents/chemistry/items.dm

```diff
@@ -0,0 +1,162 @@
+///if the pHmeter gives a detailed output
+#define DETAILED_CHEM_OUTPUT 1 
+///if the pH meter gives a shorter output
+#define SHORTENED_CHEM_OUTPUT 0 
+
+/*
+* a pH booklet that contains pH paper pages that will change color depending on the pH of the reagents datum it's attacked onto
+*/
+/obj/item/pHbooklet
+	name = "pH indicator booklet"
+	desc = "A booklet containing paper soaked in universal indicator."
+	icon_state = "pHbooklet"
+	icon = 'icons/obj/chemical.dmi'
+	item_flags = NOBLUDGEON
+	resistance_flags = FLAMMABLE
+	w_class = WEIGHT_CLASS_TINY
+	///How many pages the booklet holds
+	var/number_of_pages = 50
+
+//A little janky with pockets
+/obj/item/pHbooklet/attack_hand(mob/user, act_intent = user.a_intent, unarmed_attack_flags)
+	if(user.get_held_index_of_item(src))//Does this check pockets too..?
+		if(number_of_pages == 50)
+			icon_state = "pHbooklet_open"
+		if(number_of_pages >= 1)
+			var/obj/item/pHpaper/P = new /obj/item/pHpaper
+			P.add_fingerprint(user)
+			P.forceMove(user.loc)
+			user.put_in_active_hand(P)
+			to_chat(user, "<span class='notice'>You take [P] out of \the [src].</span>")
+			number_of_pages--
+			playsound(user.loc, 'sound/items/poster_ripped.ogg', 50, TRUE)
+			add_fingerprint(user)
+			if(number_of_pages == 0)
+				icon_state = "pHbooklet_empty"
+			return
+		else
+			to_chat(user, "<span class='warning'>[src] is empty!</span>")
+			add_fingerprint(user)
+			return
```

Comment:
```suggestion
		if(!number_of_pages)
			to_chat(user, "<span class='warning'>[src] is empty!</span>")
			add_fingerprint(user)
			return
		var/obj/item/pHpaper/page = new(get_turf(user))
		page.add_fingerprint(user)
		user.put_in_active_hand(page)
		to_chat(user, "<span class='notice'>You take [page] out of \the [src].</span>")
		number_of_pages--
		playsound(user.loc, 'sound/items/poster_ripped.ogg', 50, TRUE)
		add_fingerprint(user)
		if(number_of_pages == 0)
			icon_state = "pHbooklet_empty"
		return
```
Early return and minor things.

Reply:
This was one of the first things I wrote. It really shows. I should've thought to go over it, but thanks for pointing it out in detail.